ui/learn/src/map/mapSide.ts

```diff
@@ -89,19 +90,76 @@ function renderHome(ctrl: SideCtrl) {
         },
       }),
     ]),
-    m('div.actions', [
+    m(
+      'div.actions',
       progress > 0
-        ? m(
-            'a.confirm',
-            {
-              onclick: function () {
-                if (confirm(ctrl.trans.noarg('youWillLoseAllYourProgress'))) ctrl.reset();
+        ? [
+            m(
+              'label',
+              {
+                for: 'reset-categ-dropdown',
               },
-            },
-            ctrl.trans.noarg('resetMyProgress')
-          )
-        : null,
-    ]),
+              ctrl.trans.noarg('resetMyProgress')
+            ),
+            m(
+              'select',
+              {
+                name: 'reset-categ-dropdown',
+                id: 'reset-categ-dropdown',
+                oninput: (value: Event | null) => {
+                  const target = value?.target as HTMLSelectElement;
+                  const question = target.selectedOptions[0].getAttribute('question');
+                  if (question) {
```

Comment:
Is this even necesssary? For the "choose" case," the value won't be set anyways so the if below will fail?

Reply:
Nop because there is no "value", and also no "question" (double validation) for the "Choose" case.